### PR TITLE
testing/efl: upgrade to 1.23.0

### PR DESCRIPTION
**Pull request by billiob**
Build system has changed to meson/ninja

### CODE REVIEW

**Comment by billiob** [comment]
I forgot to disable nls. Will update PR once I've tested that.

File: testing/efl/APKBUILD [hunk 1]
```diff
@@ -13,55 +13,100 @@ depends_dev="giflib-dev gstreamer-dev gst-plugins-base-dev libpng-dev
...
+	freetype-dev \
+	fribidi-dev \
+	harfbuzz-dev \
+	libjpeg-turbo-dev \
+	libpng-dev \
+	libsndfile-dev \
+	libx11-dev \
+	libxkbcommon-dev \
+	openssl-dev \
+	wayland-dev"
```
Comment (Cogitri) [inline testing/efl/APKBUILD]: No need for the backslash

File: testing/efl/APKBUILD [hunk 2]
```diff
@@ -13,55 +13,100 @@ depends_dev="giflib-dev gstreamer-dev gst-plugins-base-dev libpng-dev
...
 options="!check" # Test suite issues
 subpackages="$pkgname-dev"
-source="https://download.enlightenment.org/rel/libs/efl/efl-$pkgver.tar.xz"
+source="https://download.enlightenment.org/rel/libs/$pkgname/$pkgname-$pkgver.tar.xz"
+builddir="$srcdir/$pkgname-$pkgver"
```
Comment (Cogitri) [inline testing/efl/APKBUILD]: This is the default already, so please remove it.

File: testing/efl/APKBUILD [hunk 3]
```diff
@@ -13,55 +13,100 @@ depends_dev="giflib-dev gstreamer-dev gst-plugins-base-dev libpng-dev
...
+	libpng-dev \
+	libsndfile-dev \
+	libx11-dev \
+	libxkbcommon-dev \
+	openssl-dev \
+	wayland-dev"
 options="!check" # Test suite issues
 subpackages="$pkgname-dev"
-source="https://download.enlightenment.org/rel/libs/efl/efl-$pkgver.tar.xz"
+source="https://download.enlightenment.org/rel/libs/$pkgname/$pkgname-$pkgver.tar.xz"
```
Comment (Cogitri) [inline testing/efl/APKBUILD]: Please don't use $pkgname in source

File: testing/efl/APKBUILD [hunk 4]
```diff
@@ -13,55 +13,100 @@ depends_dev="giflib-dev gstreamer-dev gst-plugins-base-dev libpng-dev
...
-		--with-dbus-services= \
-		--with-glib=no \
-		--with-js=none \
-		--with-opengl=none \
-		--with-x \
-		--with-x11=xlib
-	make
+		--libdir /usr/lib \
+		--buildtype=release \
+		--default-library shared \
```
Comment (Cogitri) [inline testing/efl/APKBUILD]: I think you're missing a `=` here

File: testing/efl/APKBUILD [hunk 5]
```diff
@@ -13,55 +13,100 @@ depends_dev="giflib-dev gstreamer-dev gst-plugins-base-dev libpng-dev
...
+		 -Dbuild-tests=false \
+		 -Decore-imf-loaders-disabler=ibus,scim \
+		 . build
+
+	 ninja -C build
 }
 
 package() {
-	make DESTDIR="$pkgdir" install
+	DESTDIR="$pkgdir" ninja -C "$builddir"/build install
```
Comment (Cogitri) [inline testing/efl/APKBUILD]: No need for the `$builddir` here.

File: testing/efl/APKBUILD [hunk 6]
```diff
@@ -13,55 +13,100 @@ depends_dev="giflib-dev gstreamer-dev gst-plugins-base-dev libpng-dev
...
+	libxrandr \
+	libxtst \
+	libxscrnsaver \
+	libxkbcommon \
+	libsndfile \
+	librsvg \
+	mesa-gles \
+	luajit \
+	wayland \
+	openssl"
```
Comment (Cogitri) [inline testing/efl/APKBUILD]: Most if not all of these are autodetected by abuild. Please see the build output for that.

File: testing/efl/APKBUILD [hunk 7]
```diff
@@ -13,55 +13,100 @@ depends_dev="giflib-dev gstreamer-dev gst-plugins-base-dev libpng-dev
...
+	libxcursor-dev \
+	libxdamage-dev \
+	libxinerama-dev \
+	libxrandr-dev \
+	libxtst-dev \
+	libxscrnsaver-dev \
+	mesa-dev \
+	xorgproto \
+	luajit-dev \
+	wayland-protocols"
```
Comment (Cogitri) [inline testing/efl/APKBUILD]: Same here.

**Comment by billiob** [comment]
About `depends_dev="luajit-dev"`, this is a known bug in apk. See https://gitlab.alpinelinux.org/alpine/aports/issues/10089

File: testing/efl/APKBUILD [hunk 8]
```diff
@@ -13,55 +13,74 @@ depends_dev="giflib-dev gstreamer-dev gst-plugins-base-dev libpng-dev
 	libxscrnsaver-dev xorgproto luajit-dev"
 makedepends="$depends_dev dbus-dev openssl-dev fontconfig-dev freetype-dev bash
 	fribidi-dev zlib-dev"
+depends=""
+depends_dev="luajit-dev"
+makedepends="$depends_dev
```
Comment (Cogitri) [inline testing/efl/APKBUILD]: makedepends and depends_dev is already defined two lines earlier, please don't define it twice but append to the previous variables instead.

Comment (billiob) [reply testing/efl/APKBUILD]: I can really be a moron sometimes. Sorry

Comment (Cogitri) [reply testing/efl/APKBUILD]: No worries

File: testing/efl/APKBUILD [hunk 9]
```diff
@@ -1,67 +1,80 @@
...
 url="https://www.enlightenment.org"
 arch="all !s390x" #luajit not available
 license="LGPL-2.1-only"
-depends_dev="giflib-dev gstreamer-dev gst-plugins-base-dev libpng-dev
-	libjpeg-turbo-dev libx11-dev libxcomposite-dev libxcursor-dev
-	libxdamage-dev libxinerama-dev libxrandr-dev libxtst-dev
-	libxscrnsaver-dev xorgproto luajit-dev"
-makedepends="$depends_dev dbus-dev openssl-dev fontconfig-dev freetype-dev bash
-	fribidi-dev zlib-dev"
+depends=""
```
Comment (Cogitri) [inline testing/efl/APKBUILD]: No need to define an empty variable, just leave this out :)

**Pull request closed without merging**